### ISSUE TITLE
Dolt handles NOW() incorrectly

### ISSUE THREAD

**Issue by zachmu** [dolthub/dolt#871]
Dolt seems to be reusing the session start time for successive values of NOW().

```
timestamps> create table a (x int primary key not null, t timestamp);
timestamps> insert into a values (2, now());
Query OK, 1 row affected
timestamps> insert into a values (3, now());
Query OK, 1 row affected
timestamps> insert into a values (4, now());
Query OK, 1 row affected
timestamps> insert into a values (4, now());
Query OK, 1 row affected
timestamps> select * from a;
+---+---------------------------------------+
| x | t                                     |
+---+---------------------------------------+
| 1 | 2020-09-11 21:53:24.7273339 +0000 UTC |
| 2 | 2020-09-11 21:57:14.3523038 +0000 UTC |
| 3 | 2020-09-11 21:57:14.3523038 +0000 UTC |
| 4 | 2020-09-11 21:57:14.3523038 +0000 UTC |
+---+---------------------------------------+
```

**Comment by reltuk** [comment]
AFAIK, `NOW()` is supposed to return the same value throughout the query, and should return different values for each new query. I'm not sure if in MySQL, across a single query, it returns the value close to the time when the query started, or if it returns the value close to the time it was first evaluated within the query.

In `go-mysql-server`, it's returning the value for `sql.Context`'s `queryTime` field. I think we used to create a new `sql.Context` every query, but when we moved some per-session state onto the `Context` we changed that behavior. We probably need to change our entry points so that we get a new `sql.Context` value with an updated `queryTime` appropriately when we're starting the processing of a new query.
